core/src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsTests.java

```diff
@@ -941,6 +950,110 @@ public void testNodeNotReachableFromMaster() throws Exception {
         ensureStableCluster(3);
     }
 
+    @Test
+    public void testIndexImportedFromDataOnlyNodesIfMasterLostDataFolder() throws Exception {
+        // test for https://github.com/elastic/elasticsearch/issues/8823
+        configureMulticastCluster(3, 1);
+        String masterNode = startMasterNode();
+        String blueNode = startDataNode("blue");
+        String redNode = startDataNode("red");
+
+        // create blue_index on blue_node and same for red
+        client().admin().cluster().health(clusterHealthRequest().waitForYellowStatus().waitForNodes("3")).get();
+        assertAcked(prepareCreate("blue_index").setSettings(Settings.builder().put("index.number_of_replicas", 0).put(FilterAllocationDecider.INDEX_ROUTING_INCLUDE_GROUP + "color", "blue")));
+        index("blue_index", "doc", "1", jsonBuilder().startObject().field("text", "some text").endObject());
+        assertAcked(prepareCreate("red_index").setSettings(Settings.builder().put("index.number_of_replicas", 0).put(FilterAllocationDecider.INDEX_ROUTING_INCLUDE_GROUP + "color", "red")));
+        index("red_index", "doc", "1", jsonBuilder().startObject().field("text", "some text").endObject());
```

Comment:
discussed and agreed to move to one index, in which case we don't need the colors.